bin/tx-prover/src/proxy/mod.rs

```diff
@@ -27,6 +33,7 @@ const WORKER_CHECK_INTERVAL_MILLIS: Duration = Duration::from_millis(20);
 /// Load balancer that uses a round robin strategy
 pub struct LoadBalancer {
     available_workers: Arc<RwLock<Vec<Backend>>>,
+    current_workers: Arc<RwLock<Vec<Backend>>>,
```

Comment:
Another approach could be to have a single list of workers where each worker also tracks its on status. For example, something like:
```rust
pub struct LoadBalancer {
    workers: Arc<RwLock<Vec<Worker>>,
    ...
}

struct Worker {
    backend: Backend,
    is_available: bool,
}
```
One downside of this approach is that we'd need to iterate through the vector to find the next available worker, but given that we probably won't have more than few hundred workers, I don't think that's an issue.

At the same time, this may simplify reasoning about logic related to handling worker insertions and removals.

Reply:
This approach is indeed better, also prevents deadlocks in comparison with the previous approach.